### ISSUE TITLE
Default format time series xtick labels changed

### ISSUE THREAD

**Issue by Puggie** [matplotlib/matplotlib#6365]
The default format to display time series xtick labels in time series plots seems to have been changed between matplotlib 1.4 and 1.5. When using time stamps in a datetime format with a 1 second resolution and plotting a minute worth of data, the major xtick labels are now formatted as "HH:MM:SS.mmmmm" as opposed to "HH:MM:SS" in matplotlib 1.4. This new behaviour doesn't match the significany of the timestamps and results in overlapping xtick labels as well.

A simple example to illustrate the issue:

```
import numpy as np
import matplotlib.pyplot as plt
import datetime

# generating time stamps with a resolution of 1s
x = np.array([datetime.datetime(2016, 5, 4, 12, 0, i) for i in range(60)])
# generating some random y-values
y = np.random.randn(x.shape[0]).cumsum()

# plot
plt.plot(x, y)
plt.show()
```
 This results in the following figure:
![image](https://cloud.githubusercontent.com/assets/3796187/15014847/8f482f6a-1208-11e6-8152-4999e6a7b013.png)

Of course, the xtick label format can be manually adjusted:
```
import matplotlib.dates as dates

# create plot
plt.plot(x, y)
# get reference to x-axis
xax = plt.gca().get_xaxis()
# format major xtick label
xax.set_major_formatter(dates.DateFormatter('%H:%M:%S'))
# show plot
plt.show()
```
But I found the old behavior more elegant.

I'm using matplotlib 1.5.1 on Python 3.5.1 32-bit as part of the Anaconda3 4.0.0 Windows x86 distrubtion. The OS is Windows 7 64 bit.

**Comment by tacaswell** [comment]
What version did you have the "HH:MM:SS" behavior?  I can not reproduce that format on 1.4.3 or 1.5.x.

The formatter used is based on the range of the data (not the resolution), hence 


```python
import numpy as np
import matplotlib.pyplot as plt
import datetime

# generating time stamps with a resolution of 1s
x = np.array([datetime.datetime(2016, 5, 4, 12, j, i) for j in range(60) for i in range(60)])
# generating some random y-values
y = np.random.randn(x.shape[0]).cumsum()

# plot
plt.plot(x, y)
plt.show()
```

Gives 

![so](https://cloud.githubusercontent.com/assets/199813/15135389/a7be6b60-1643-11e6-8c7e-9e4852c25082.png)

In 2.x the formats for each of the 'default' break points will be configurable by rcParams.

Can you verify that you get the expected old behavior if you down grade mpl (conda has packages for 1.4.x and 1.3.x)?

**Comment by Puggie** [comment]
I created a conda evironment with python 3.4.2 and matplotlib 1.4.3 (which was an older configuration I once used). It appears that in both this environment and the environment mentioned in my original post, I get the xtick labels in "HH:MM:SS.mmmmm" format when running either your or mine code example.
The difference in the tick label format between matplotlib versions actually occured in my production code, which is not suitable for posting. So it is somewhat difficult for me to prove the difference in tick label format here. What remains is that the tick label format is still deviating, but then for both an older and a recent matplotlib version.

**Comment by tacaswell** [comment]
We are sorry about the changes to your production code, but without a reproducible example it is very hard to do anything about this.  If you can not produce a minimal example, knowing what your inputs look like, we have no chance.

Did you have pandas imported or not?  They install different date unit handlers.

**Comment by Puggie** [comment]
In my production code, I did import pandas indeed. So I ran the following code in my old environment (with pandas version 0.16.2):
```
import numpy as np
import matplotlib.pyplot as plt
import datetime
import pandas as pd

# generating time stamps with a resolution of 1s
x = np.array([datetime.datetime(2016, 5, 4, 12, j, i) for j in range(60) for i in range(60)])
# generating some random y-values
y = np.random.randn(x.shape[0]).cumsum()

# plot
plt.plot(x, y)
plt.show()
```
This gave me the old behavior with ""HH:MM:SS" format xtick labels. Running the same code in my new environment (pandas 0.18.0) gave me the "HH:MM:SS.mmmmm" formatted labels again.

**Comment by tacaswell** [comment]
attn @jreback who on the pandas side would know about the datetime unit handler code.

**Comment by Puggie** [comment]
Just to reassure my conda environment was not somehow messed up: I just created a fresh new conda environment (python 3.5)  with only matplotlib 1.5.1 and its dependencies (no pandas!). When running the code without the line to import pandas, I still get the ""HH:MM:SS.mmmmm" formatted labels.

The following command was used to create the environment:
`conda create --name MPLnewonly python=3.5 matplotlib`
This environment contains the following packages:
```
[MPLnewonly] C:\Anaconda3\envs\MPLnewonly>conda list
# packages in environment at C:\Anaconda3\envs\MPLnewonly:
#
cycler                    0.10.0                   py35_0    defaults
jpeg                      8d                       vc14_0  [vc14]  <unknown>
libpng                    1.6.17                   vc14_1  [vc14]  <unknown>
libtiff                   4.0.6                    vc14_1  [vc14]  defaults
matplotlib                1.5.1               np111py35_0    defaults
mkl                       11.3.1                        0    defaults
numpy                     1.11.0                   py35_0    defaults
openssl                   1.0.2h                   vc14_0  [vc14]  defaults
pip                       8.1.1                    py35_1    defaults
pyparsing                 2.1.1                    py35_0    defaults
pyqt                      4.11.4                   py35_5    defaults
python                    3.5.1                         4    defaults
python-dateutil           2.5.2                    py35_0    defaults
pytz                      2016.4                   py35_0    defaults
qt                        4.8.7                    vc14_7  [vc14]  defaults
setuptools                20.7.0                   py35_0    defaults
sip                       4.16.9                   py35_2    defaults
six                       1.10.0                   py35_0    <unknown>
tk                        8.5.18                   vc14_0  [vc14]  <unknown>
vs2015_runtime            14.00.23026.0                 0    defaults
wheel                     0.29.0                   py35_0    defaults
zlib                      1.2.8                    vc14_2  [vc14]  <unknown>
```
So it seems the issue isn't caused by pandas.

**Comment by tacaswell** [comment]
The fact that you were seeing "HH:MM:SS" is because of pandas in 1.4.3 (as if you do not import pandas in 1.4.3 you see the format with the decimal places). 

It looks like this was due to a change in pandas (from me) https://github.com/pydata/pandas/pull/11770

Downgrade pandas with mpl 1.5.1 and I bet it works.

**Comment by jreback** [comment]
@tacaswell best would be 

cc @TomAugspurger 
cc @sinhrks

**Comment by tacaswell** [comment]
but in this case I broke pandas :disappointed: .

**Comment by Puggie** [comment]
mpl 1.5.1 combined with pandas downgraded to 0.17.1 indeed gave me back the old behavior ("HH:MM:SS") when importing pandas.

I'm still a bit confused:
- With no involvement of pandas at all (pandas __not__ imported), the behavior is "HH:MM:SS.mmmmm" (occurs with both older and new versions of pandas  installed and with pandas not installed at all). So that makes me think this is the intended behavior of mpl. Is that correct?
- But then again: why is it that this behavior is different between the installation of @tacaswell (resulting in "HH:MM:SS") and mine (resulting in "HH:MM:SS.mmmmm")?

**Comment by tacaswell** [comment]
in mpl < 2.0 the scales -> format string mapping is hard-coded.  We have not changed them out of an abundance of caution about breaking API, they are being changed in 2.0 (and there will probably be another PR into 2.x as a result of this issue).  When you import pandas it replaces the mpl unit handlers for dates (which is how mpl knows how to plot datetime objects) with a pandas version (which among other things knows how to deal with np.datetime64).

My image is from a 2.x installation which has slightly different break points than 1.5.1.  I also increased the range.

**Comment by tacaswell** [comment]
addressed by #6409 and https://github.com/pydata/pandas/pull/13131
